milabench/cli/covalent/__main__.py

```diff
@@ -0,0 +1,233 @@
+import argparse
+import asyncio
+import os
+import pathlib
+import subprocess
+import sys
+import tempfile
+
+
+def serve(*argv):
+    return subprocess.run([
+        str(pathlib.Path(sys.executable).with_name("covalent")),
+        *argv
+    ]).returncode
+
+
+def _get_executor_kwargs(args):
+    return {
+        **{k:v for k,v in vars(args).items() if k not in ("setup", "teardown")},
+        **{"action":k for k,v in vars(args).items() if k in ("setup", "teardown") and v},
+    }
+
+
+def executor(executor_cls, args, *argv):
+    import covalent as ct
+
+    executor:ct.executor.BaseExecutor = executor_cls(
+        **_get_executor_kwargs(args),
+    )
+
+    def _popen(cmd, *args, _env=None, **kwargs):
```

Comment:
This should probably be simplified, I don't think we'll use the complex features on configs and args paths